src/validation_workflow/docker/validation_docker.py

```diff
@@ -29,23 +26,57 @@ def __init__(self, args: ValidationArgs) -> None:
         super().__init__(args)
 
     def download_artifacts(self) -> bool:
-
-        # Check if the docker daemon is running; or else, bring it up
-        if (self.is_container_daemon_running()):
-
-            # STEP 1 . pull the images for OS and OSD
-            self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
-            self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
```

Comment:
Thanks for the comment, @dblock , the new code hopefully explains more on selecting the image source between dockerhub and ecr, and plus supporting staging and production...
the code will choose either dockerhub or ecr ; and either using staging-only or comparing between staging and production before briing up a container, and it will be based on the arguemnts taken from the user input

sorry that I am lost when you comment on 'pass the image name dynamically' ... I guess you don't mean to have user to input the full image name as an argument if I am not mistaken .. thanks,

Reply:
Thanks, @dblock ..I made an update just now .. hopefully it will work betteer this time..